File: src/Plotly.NET/Caps.fs
namespace Plotly.NET

open DynamicObj
open System

type CapFill () =
    inherit DynamicObj () 

    static member init 
        (
            ?Fill: float,
            ?Show: bool
        ) =
            CapFill()
            |> CapFill.style
                (
                    ?Fill = Fill,
                    ?Show = Show
                )

    static member style
        (
            ?Fill: float,
            ?Show: bool
        ) =

            fun (capFill: CapFill) ->
                
                Fill |> DynObj.setValueOpt capFill "fill"
                Show |> DynObj.setValueOpt capFill "show"


type Caps() =
    inherit DynamicObj ()

    static member init
        (
            ?X: CapFill,
            ?Y: CapFill,
            ?Z: CapFill
        ) =

            Caps()
            |> Caps.style
                (
                    ?X  = X,
                    ?Y  = Y,
                    ?Z  = Z
                )
            
    static member style 
        (
            ?X: CapFill,
            ?Y: CapFill,
            ?Z: CapFill
        ) =
            fun (caps: Caps) ->

                X   |> DynObj.setValueOpt caps "x"
                Y   |> DynObj.setValueOpt caps "y"
                Z   |> DynObj.setValueOpt caps "z"

                caps

File: src/Plotly.NET.CSharp/ObjectAbstractions/TraceObjects/Caps.cs

using Microsoft.FSharp.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Plotly.NET.CSharp;

public static partial class TraceObjects
{
    public static Plotly.NET.TraceObjects.CapFill CapFill
        (

            Optional<double> Fill = default,
            Optional<bool> Show = default
        ) => 
            Plotly.NET.TraceObjects.CapFill.init(

                Fill: Fill.ToOption(),
                Show: Show.ToOption()
            );
}

public static partial class TraceObjects
{
    public static Plotly.NET.TraceObjects.Caps Caps
        (

            Optional<Plotly.NET.TraceObjects.CapFill> X = default,
            Optional<Plotly.NET.TraceObjects.CapFill> Y = default,
            Optional<Plotly.NET.TraceObjects.CapFill> Z = default
        ) => 
            Plotly.NET.TraceObjects.Caps.init(

                X: X.ToOption(),
                Y: Y.ToOption(),
                Z: Z.ToOption()
            );
}

